go/vt/mysqlctl/xtrabackupengine.go

```diff
@@ -0,0 +1,401 @@
+/*
+Copyright 2019 The Vitess Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package mysqlctl
+
+import (
+	"bufio"
+	"context"
+	"errors"
+	"flag"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"path"
+	"strings"
+
+	"github.com/klauspost/pgzip"
+	"vitess.io/vitess/go/mysql"
+	"vitess.io/vitess/go/vt/dbconfigs"
+	"vitess.io/vitess/go/vt/logutil"
+	"vitess.io/vitess/go/vt/mysqlctl/backupstorage"
+	"vitess.io/vitess/go/vt/vterrors"
+)
+
+// XtrabackupEngine encapsulates the logic of the xtrabackup engine
+// it implements the BackupEngine interface and contains all the logic
+// required to implement a backup/restore by invoking xtrabackup with
+// the appropriate parameters
+type XtrabackupEngine struct {
+}
+
+var (
+	// path where backup engine program is located
+	// TODO assume lives on path and remove this option?
+	xtrabackupEnginePath = flag.String("xtrabackup_root_path", "", "directory location of the xtrabackup executable, e.g., /usr/bin")
```

Comment:
I think this makes sense. I would leave it as is.

Reply:
why not just have path to the actual executable, rather than the parent dir?